src/AISmart.GAgent.Core/GAgentBase.cs

```diff
@@ -208,6 +208,25 @@ private async Task PublishAsync<T>(EventWrapper<T> eventWrapper) where T : Event
             return;
         }
 
+        var contextStorageGrain = eventWrapper.ContextGrainId == null
+            ? GrainFactory.GetGrain<IContextStorageGrain>(Guid.NewGuid())
+            : GrainFactory.GetGrain<IContextStorageGrain>(eventWrapper.ContextGrainId.Value);
+        eventWrapper.ContextGrainId = contextStorageGrain.GetGrainId();
+
+        foreach (var keyPair in eventWrapper.Event.GetContext())
+        {
+            await contextStorageGrain.AddContextAsync(keyPair.Key, keyPair.Value);
```

Comment:
Think we call AddContextAsync just once, so we pass in the whole context instead of adding key value one by one since grain calls can happen over the net